src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,88 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private int size;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        this.size = 0;
+        this.keys = (K[]) new Object[MAX_ITEMS_NUMBER];
+        this.values = (V[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (key == null && value != null) {
+            int firstKeyNullIndex = this.getFirstKeyNullIndex();//find first index with key == null
+            if (firstKeyNullIndex == -1 && size == 10) {
+                //нет ни одного key==null, значит все key уже заполнены
+                throw new RuntimeException("There are not empty elements in the storage");
+            } else {
+                if (values[firstKeyNullIndex] == null) {
+                    this.values[firstKeyNullIndex] = value;//add value to this index
+                    this.size++;
+                } else {
+                    this.values[firstKeyNullIndex] = value;//rewrite value at this index
+                }
+            }
+        } else {
+            int index = this.getKeyIndex(key);
+            if (index == -1) {
+                this.keys[this.size] = key;
+                this.values[this.size] = value;
+                this.size++;
+            } else {
+                this.values[index] = value;
+            }
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        if (key == null) {
+            int firstKeyNullIndex = this.getFirstKeyNullIndex();
+            return values[firstKeyNullIndex];
+        } else {
+            int i = -1;
+            for (K element : keys) {
+                i++;
+                if (element != null && element.equals(key)) {
+                    return values[i];
+                }
+            }
+            return null;//прийдется вернуть null хоть это и bad practice но так требуют тесты
+        }
     }
 
     @Override
     public int size() {
+        return this.size;
```

Comment:
```suggestion
        return size;
```

нам не нужно тут ключевое слово `this`